src/torchcodec/decoders/_video_decoder.py

```diff
@@ -273,14 +282,23 @@ def get_frames_displayed_at(
 
 def _get_and_validate_stream_metadata(
     decoder: Tensor,
+    stream_index: Optional[int] = None,
 ) -> Tuple[core.VideoStreamMetadata, int]:
     video_metadata = core.get_video_metadata(decoder)
 
     best_stream_index = video_metadata.best_video_stream_index
-    if best_stream_index is None:
+    if best_stream_index is None and stream_index is None:
         raise ValueError(
-            "The best video stream is unknown. " + _ERROR_REPORTING_INSTRUCTIONS
+            "The best video stream is unknown and there is no specified stream. "
+            + _ERROR_REPORTING_INSTRUCTIONS
         )
```

Comment:
Nit
We have 2 instances of `if stream_index` is None, and we're retrieving `best_stream_index` unconditionally while it should only be needed when `streaming_index` is None. I'd suggest to move this block within a single `if stream_index is None` below, starting at line 296